agenda/network/src/main/java/com/pronoidsoftware/agenda/network/KtorRemoteAgendaDataSource.kt

```diff
@@ -133,32 +149,75 @@ class KtorRemoteAgendaDataSource @Inject constructor(
-                            val photoName = "photo$index"
-                            append(
-                                photoName,
-                                URL(localPhotoUri).readBytes(),
-                                // TODO: get bytes of created compressed file
-                                Headers.build {
-                                    append(HttpHeaders.ContentType, "image/png")
-                                    append(
-                                        HttpHeaders.ContentDisposition,
-                                        "filename=$photoName.png",
-                                    )
-                                },
-                            )
-                        }
-                },
-            ),
-        ).map { it.toEvent(sessionStorage.get()?.userId) }
+        val compressRequest = OneTimeWorkRequestBuilder<CompressPhotosWorker>()
+            .setInputData(
+                workDataOf(
+                    "KEY_URIS_TO_COMPRESS" to
+                        event.photos
+                            .filterIsInstance<Photo.Local>()
+                            .map { it.localPhotoUri }
+                            .toTypedArray(),
+                    "KEY_COMPRESSION_THRESHOLD" to 1_048_576L,
+
+                ),
+            )
+            .build()
+        val createEventWorkRequest = OneTimeWorkRequestBuilder<CreateEventWorker>()
+            .setConstraints(
+                Constraints.Builder()
+                    .setRequiredNetworkType(
+                        NetworkType.CONNECTED,
+                    )
+                    .build(),
+            )
+            .setInputData(
+                workDataOf(
+                    "KEY_CREATE_EVENT_REQUEST" to Json.encodeToString(event.toCreateEventRequest()),
+                ),
+            )
+            .build()
+        val workManager = WorkManager.getInstance(context)
+        workManager
+            .beginUniqueWork(
+                "createEvent",
+                ExistingWorkPolicy.REPLACE,
+                compressRequest,
+            )
+            .then(createEventWorkRequest)
+            .enqueue()
+            .await()
+        val workInfos = withContext(dispatchers.io) {
+            workManager.getWorkInfosForUniqueWork("createEvent").get()
+        }
+        val createInfo = workInfos.find { it.id == createEventWorkRequest.id }
+        return when (createInfo?.state) {
+            WorkInfo.State.SUCCEEDED -> {
+                val skippedPhotoCount = createInfo.outputData.getInt("SKIPPED_PHOTO_COUNT", 0)
+                if (skippedPhotoCount > 0) {
+                    Toast
+                        .makeText(
+                            context,
+                            "$skippedPhotoCount photos were skipped because they were too large",
+                            Toast.LENGTH_LONG,
+                        )
+                        .show()
+                }
```

Comment:
Even though this works, I would consider showing a toast a responsibility of the UI. Doing it here in the repo could show at any point, even if the user isn't even in the app anymore, so they lack context when this toast shows.